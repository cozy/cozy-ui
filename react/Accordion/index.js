import Accordion from '@material-ui/core/Accordion'

export default Accordion
