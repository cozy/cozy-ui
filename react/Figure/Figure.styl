@require 'settings/breakpoints'


&.Figure-currency
    color var(--secondaryTextColor)

&.Figure-content--positive
    color var(--successColor)

    .Figure-currency
        color var(--successColor)

&.Figure-content--negative
    color var(--errorColor)

    .Figure-currency
        color var(--errorColor)

&.Figure-content--warning
    color var(--warningColor)

    .Figure-currency
        color var(--warningColor)

.Figure-total
    font-weight    900

.Figure__currency--withSpacing
    margin-left .2em

.Figure--big
    font-size 2rem
    line-height 2.625rem

+small-screen()
    .Figure--big
        font-size 1.5rem
        line-height 1.75rem

.Figure--clickable
    cursor pointer

.Figure_blur
    filter blur(8px)

.Figure--inline
    display inline
