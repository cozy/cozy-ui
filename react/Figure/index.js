export { default, default as Figure } from './Figure'
export { default as FigureBlock } from './FigureBlock'
