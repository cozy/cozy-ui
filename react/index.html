<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Cozy UI React components</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Inter:400,500,600">
    <link rel="stylesheet" href="main.css">
  </head>
  <body>
    <div id="rsg-root"></div>
    <script src="build/bundle.3c2f80ae.js"></script>
  </body>
</html>