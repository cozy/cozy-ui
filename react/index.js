export {default as Alerter} from './Alerter'
export { Button, ButtonLink } from './Button'
export {default as I18n, translate} from './I18n'
export {default as Icon} from './Icon'
export {default as Sidebar} from './Sidebar'
export {default as Nav, NavItem, NavIcon, NavText, NavLink} from './Nav'
export {default as Modal, ModalContent, ModalSection, ModalButtons} from './Modal'
export {default as SelectionBar} from './SelectionBar'
export {default as IntentOpener} from './IntentOpener'
export {default as Spinner} from './Spinner'
export {Tabs, TabPanels, TabPanel, TabList, Tab} from './Tabs'
export {default as Toggle} from './Toggle'
export {default as withBreakpoints} from './helpers/withBreakpoints'
export { Media, Img, Bd } from './Media'
export { default as ActionMenu } from './ActionMenu'
export { default as Menu, MenuItem } from './Menu'
export { default as Overlay } from './Overlay'
