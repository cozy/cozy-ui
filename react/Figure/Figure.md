```jsx
import { Figure } from 'cozy-ui/transpiled/react/Figure';
<div>
  <Figure
    total={1000}
    symbol='€'
    coloredPositive coloredNegative signed />

  <Figure
    total={-1000}
    symbol='€'
    coloredPositive coloredNegative signed />

  <Figure
    total={-1000}
    symbol='€'
    signed />
</div>
```
