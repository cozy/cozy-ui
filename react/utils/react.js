import { Fragment, Children } from 'react'

export const getComponentName = component => {
  if (!component) return null
  return component.type.displayName || component.type.name || 'Component'
}

export const getChildren = props => {
  return Children.toArray(props.children)
}

export const getChildrenCount = props => {
  const children = getChildren(props)
  return children.length
}

export const getLastChild = props => {
  const children = getChildren(props)
  const lastChild = children[children.length - 1]

  return lastChild.type === Fragment
    ? getChildren(lastChild.props)[0]
    : lastChild
}
