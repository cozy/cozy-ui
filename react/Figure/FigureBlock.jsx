import React from 'react'
import Types from 'prop-types'
import classNames from 'classnames'
import Figure from './Figure'
import Typography from '../Typography'
import styles from './FigureBlock.styl'

/**
 * Shows a `Figure` with a label, useful for important numbers.
 *
 * A part from `className` and `label`, takes same properties
 * as `Figure`.
 */
const FigureBlock = ({
  className,
  label,
  total,
  symbol,
  coloredPositive,
  coloredNegative,
  signed,
  decimalNumbers = 0,
  figureClassName,
  withCurrencySpacing
}) => (
  <div className={classNames(styles['FigureBlock'], className)}>
    <Typography variant="subtitle1" color="textSecondary">
      {label}
    </Typography>
    <Figure
      size="big"
      className={classNames(styles['FigureBlock-figure'], figureClassName)}
      total={total}
      symbol={symbol}
      coloredPositive={coloredPositive}
      coloredNegative={coloredNegative}
      signed={signed}
      decimalNumbers={decimalNumbers}
      withCurrencySpacing={withCurrencySpacing}
    />
  </div>
)

FigureBlock.propTypes = {
  /** Label of the figure */
  label: Types.string.isRequired,
  /** Extra classname */
  className: Types.string
}

export default FigureBlock
