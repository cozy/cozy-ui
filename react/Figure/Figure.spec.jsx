import { render } from '@testing-library/react'
import { flatten, merge } from 'lodash'
import React from 'react'

import Figure from './Figure'

const combine = (...iterables) => {
  if (iterables.length === 1) {
    return iterables[0].map(x => [x])
  } else {
    const combinationsNMinus1 = combine.apply(null, iterables.slice(1))
    return flatten(
      combinationsNMinus1.map(c => iterables[0].map(item => [item, ...c]))
    )
  }
}

const formatAttrs = attrs => {
  return Object.keys(attrs)
    .map(x => `${x}: ${attrs[x]}`)
    .join(', ')
}

describe('Figure', () => {
  const amounts = [-100, 100, 500, 4]

  const coloredAttributes = [
    'coloredPositive',
    'coloredNegative',
    'coloredWarning'
  ]

  const combinations = combine
    .apply(
      null,
      coloredAttributes.map(x => [{ [x]: true }, { [x]: false }])
    )
    .map(attrs => {
      return merge.apply(null, [{}, ...attrs])
    })

  // We check for different king of amounts if
  // - its red if it is negative if coloredNegative is set
  // - its green if it is positive if coloredPositive is set
  // - its yellow if it is above the warning limit if coloredWarning is set

  for (let amount of amounts) {
    for (let attrs of combinations) {
      it(`should render correctly ${amount} ${formatAttrs(attrs)}`, () => {
        const { container } = render(
          <Figure warningLimit={110} total={amount} {...attrs} />
        )
        expect(container).toMatchSnapshot()
      })
    }
  }
})
