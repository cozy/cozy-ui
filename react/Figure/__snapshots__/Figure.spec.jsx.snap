// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure should render correctly -100 coloredPositive: false, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: false, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--warning___1Pn6n"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: false, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--negative___3ACbA"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: false, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--warning___1Pn6n"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: true, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: true, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--warning___1Pn6n"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: true, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--negative___3ACbA"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly -100 coloredPositive: true, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--warning___1Pn6n"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    -100,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: false, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: false, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: false, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: false, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: true, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: true, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: true, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 4 coloredPositive: true, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    4,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: false, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: false, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: false, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: false, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: true, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: true, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: true, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 100 coloredPositive: true, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    100,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: false, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: false, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: false, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: false, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className=""
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: true, coloredNegative: false, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: true, coloredNegative: false, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: true, coloredNegative: true, coloredWarning: false 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;

exports[`Figure should render correctly 500 coloredPositive: true, coloredNegative: true, coloredWarning: true 1`] = `
<div
  className="Figure__Figure-content--positive___1qyd8"
>
  <span
    className="Figure__Figure-total___MZ7Xt"
  >
    500,00
  </span>
</div>
`;
